docs/REST API/rest-api.md

```diff
@@ -5585,6 +5586,49 @@ Resends a configuration profile for the specified host.
 
 `Status: 202`
 
+### Batch-modify custom OS settings(configuration profiles)
+
+Modify configuration profiles for a team. The provided list of profiles will be the active profiles for the specified team. If no team (`team_id` or `team_name`) is provided, the profiles are applied for all hosts (Fleet Free) or for hosts that are not assigned to "No team" (Fleet Premium).
+
+For Apple (macOS, iOS, iPadOS) profiles, Fleet will send only an `InstallProfile` command (edit) for all existing profiles with the same `PayloadIdentifier` (specified in the .mobileconfig file). Fleet will send a `RemoveProfile` command to hosts for all existing profiles that are not part of the list. 
+
+`POST /api/v1/fleet/configuration_profiles/batch`
+
+#### Parameters
+
+| Name      | Type   | In    | Description                                                                                                                       |
+| --------- | ------ | ----- | --------------------------------------------------------------------------------------------------------------------------------- |
+| team_id   | number | query | _Available in Fleet Premium_ The team ID to apply the configuration profiles to. Only one of `team_name` or `team_id` may be included in the request.          |
+| team_name | string | query | _Available in Fleet Premium_ The name of the team to apply the custom settings to. Only one of `team_name` or `team_id` may be included in the request. |
+| dry_run   | bool   | query | Validate the provided profiles and return any validation errors, but do not apply the changes.                                    |
+| configuration_profiles  | json   | body  | An array of objects, consisting of a `profile` Base64 encoded .mobileconfig or JSON profile for Apple (macOS, iOS, iPadOS) hosts or XML profile for Windows hosts, `labels_include_all`, `labels_include_any`, or `labels_exclude_any` array of label names, and a `display_name` (required). |
```

Comment:
```suggestion
| configuration_profiles  | object   | body  | An array of objects, consisting of a `profile` Base64 encoded .mobileconfig or JSON profile for Apple (macOS, iOS, iPadOS) hosts or XML profile for Windows hosts, `labels_include_all`, `labels_include_any`, or `labels_exclude_any` array of label names, and a `display_name` (required). |
```

TODO: @noahtalerman: Break out a separate table for this object like we do for other endpoints: https://fleetdm.com/docs/rest-api/rest-api#modify-configuration